File: hyeonwoo/SW 마에스트로/그리디/11047.py
import sys

N, K = map(int, sys.stdin.readline().split())
lst = []

for i in range(N):
    lst.append(int(sys.stdin.readline()))

lst.reverse()

count = 0
minimize = 999

for i in lst:
    if (K // i != 0):
        minimize = min(minimize, K // i)
        count += K // i
        K = K % i

print(count)

File: hyeonwoo/SW 마에스트로/그리디/11047.js
const fs = require('fs');
const [n, k, ...arr] = fs.readFileSync("nodeJS.txt").toString().trim().split(/\s/);

let cnt = 0;
let minimize = 999;

let N = parseInt(n) // 경우의 수
let K = parseInt(k) // 돈

arr.reverse()

for (let i of arr) {
    i = parseInt(i)
    if (K >= i && Math.floor(K / i) != 0) {
        minimize = Math.min(minimize, Math.round(K / i));
        cnt += Math.floor(K / i);
        // console.log(i, cnt, K, (K / i).toFixed())
        K = Math.round(K % i);
    }
}

console.log(cnt)